numpyro/mcmc.py

```diff
@@ -428,3 +441,303 @@ def single_chain_mcmc(rng, init_params):
+       Radford M. Neal
+
+    :param model: Python callable containing Pyro :mod:`~numpyro.primitives`.
+        If model is provided, `potential_fn` will be inferred using the model.
+    :param potential_fn: Python callable that computes the potential energy
+        given input parameters. The input parameters to `potential_fn` can be
+        any python collection type, provided that `init_params` argument to
+        `init_kernel` has the same type.
+    :param kinetic_fn: Python callable that returns the kinetic energy given
+        inverse mass matrix and momentum. If not provided, the default is
+        euclidean kinetic energy.
+    :param float step_size: Determines the size of a single step taken by the
+        verlet integrator while computing the trajectory using Hamiltonian
+        dynamics. If not specified, it will be set to 1.
+    :param bool adapt_step_size: A flag to decide if we want to adapt step_size
+        during warm-up phase using Dual Averaging scheme.
+    :param bool adapt_mass_matrix: A flag to decide if we want to adapt mass
+        matrix during warm-up phase using Welford scheme.
+    :param bool dense_mass:  A flag to decide if mass matrix is dense or
+        diagonal (default when ``dense_mass=False``)
+    :param float target_accept_prob: Target acceptance probability for step size
+        adaptation using Dual Averaging. Increasing this value will lead to a smaller
+        step size, hence the sampling will be slower but more robust. Default to 0.8.
+    :param float trajectory_length: Length of a MCMC trajectory for HMC. Default
+        value is :math:`2\\pi`.
+    """
+    def __init__(self,
+                 model=None,
+                 potential_fn=None,
+                 kinetic_fn=None,
+                 step_size=1.0,
+                 adapt_step_size=True,
+                 adapt_mass_matrix=True,
+                 dense_mass=False,
+                 target_accept_prob=0.8,
+                 trajectory_length=2*math.pi):
+        if not (model is None) ^ (potential_fn is None):
+            raise ValueError('Only one of `model` or `potential_fn` must be specified.')
+        self.model = model
+        self.potential_fn = potential_fn
+        self.kinetic_fn = kinetic_fn if kinetic_fn is not None else euclidean_kinetic_energy
+        self.step_size = step_size
+        self.adapt_step_size = adapt_step_size
+        self.adapt_mass_matrix = adapt_mass_matrix
+        self.dense_mass = dense_mass
+        self.target_accept_prob = target_accept_prob
+        self.trajectory_length = trajectory_length
+        self.sample_fn = None
+        self.algo = 'HMC'
+        self.max_tree_depth = 10
+
+    @copy_docs_from(MCMCKernel.init)
+    def init(self, rng, num_warmup, init_params=None, model_args=(), model_kwargs={}):
+        constrain_fn = None
+        if self.model is not None:
+            init_params_, self.potential_fn, constrain_fn = initialize_model(rng, self.model,
```

Comment:
A `rng` is missing at `hmc_init_fn`. I think we can split rng here.
```
rng, rng_init_model = random.split(rng)
init_params_, self.potential_fn, constrain_fn = initialize_model(rng_init_model, ...
```

Reply:
Good catch! Since none of our tests in `test_mcmc` pass `rng` directly to the `mcmc` function and instead only use it in `intialize_model`, I was getting full agreement. Made the change now.